### PR TITLE
feat: Add bq backend

### LINKED ISSUE #107: Support datasources beyond cloud logging
Cloud logging is not the most performant -- especially for looking at older data.   fleet-debugger should support bigquery as a datasource (since logs can be trivially synced there).

### PR DESCRIPTION
**Pull request by greghutch**
still WIP & not ready to go
Fixes #107

### CODE REVIEW

File: components/bigqueryDS.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyvaright 2021 Google LLC
```
Comment (bshi) [inline components/bigqueryDS.js:2]: Looks like bulk file header changes introduced a typo across all files.

File: components/bigqueryDS.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    return [];
+  }
+
+  async fetchVehicleLogs(vehicle_id, trip_id) {
+    let whereClause;
+    if (vehicle_id) {
+      whereClause = `json_value(labels.vehicle_id)="${vehicle_id}"`;
+    } else {
+      whereClause = `json_value(labels.trip_id) LIKE "%${trip_id}%"`;
+    }
```
Comment (bshi) [inline components/bigqueryDS.js:117]: Consider being more defensive w/ https://cloud.google.com/bigquery/docs/parameterized-queries#node.js 

```suggestion
    let whereClause, params = {};
    if (vehicle_id) {
      whereClause = `json_value(labels.vehicle_id)=@vehicleId`;
      params['vehicleId'] = `${vehicle_id}`;
    } else {
      whereClause = `json_value(labels.trip_id) LIKE @tripId`;
      params['tripId'] = `%${trip_id}%`;
    }
```

File: components/bigqueryDS.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    console.log("Loading trip logs for vehicle id", vehicle_id);
+    const trip_ids = _(vehicleLogs)
+      .map((x) => _.split(_.get(x, "labels.trip_id"), ","))
+      .flatten()
+      .uniq()
+      .compact()
+      .value();
+    const tripPromises = trip_ids.map(async (trip_id) => {
+      const createLogs = await this.bq(
+        `select * from \`${this.dataset}.create_trip\` where json_value(labels.trip_id)="${trip_id}" order by timestamp desc limit 10`
```
Comment (bshi) [inline components/bigqueryDS.js:84]: Consider being more defensive w/ https://cloud.google.com/bigquery/docs/parameterized-queries#node.js

**Review by bshi — APPROVED** [review]

File: components/bigqueryDS.js (hunk 1 above)
Comment (greghutch) [reply components/bigqueryDS.js]: fixed

File: components/bigqueryDS.js (hunk 2 above)
Comment (greghutch) [reply components/bigqueryDS.js]: switched to using biquery node library.  Much cleaner

File: components/bigqueryDS.js (hunk 3 above)
Comment (greghutch) [reply components/bigqueryDS.js]: mostly addressed with switch to bigquery node library

File: components/bigqueryDS.js [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+  async fetchTaskLogsForVehicle(delivery_vehicle_id) {
+    console.log(
+      "Loading tasks logs for deliveryVehicle id",
+      delivery_vehicle_id
+    );
+    const params = {
+      delivery_vehicle_id,
+      //startTime: this.startTime,
+      //endTime: this.endTime,
+      // XXX ignore start time for field day
```
Comment (bshi) [inline components/bigqueryDS.js:150]: Undo?  The values commented out are the command line defaults already no?

Or dump a WARNING log notifying user that the start/end time flags on command line are ignored?

File: components/bigqueryDS.js [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+    const params = {
+      startTime: this.startTime,
+      endTime: this.endTime,
+    };
+    if (vehicle_id) {
+      params.vehicle_id = vehicle_id;
+      whereClause = "labels.vehicle_id=@vehicle_id";
+      console.log(`Fetching logs for vehicle ${vehicle_id}`);
+    } else {
+      // XXX why like here?
```
Comment (bshi) [inline components/bigqueryDS.js:207]: ?

File: components/bigqueryDS.js [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+      whereClause = `labels.trip_id LIKE @trip_id"`;
+      params.trip_id = `%${trip_id}%`;
+      console.log(`Fetching logs for trip ${trip_id}`);
+    }
+
+    const createData = await this.bq(
+      `select * from \`${this.tablePrefix}create_vehicle\` where ${whereClause} and ${this.timeClause} order by timestamp desc limit 5000`,
```
Comment (bshi) [inline components/bigqueryDS.js:214]: nit: format SQL for readability everywhere as done in L155?
```suggestion
      `
SELECT * FROM \`${this.tablePrefix}create_vehicle\`
WHERE ${whereClause} and ${this.timeClause}
ORDER BY TIMESTAMP DESC LIMIT 5000`,
```

**Review by bshi — APPROVED** [review]

File: components/bigqueryDS.js (hunk 4 above)
Comment (greghutch) [reply components/bigqueryDS.js:150]: removed.

File: components/bigqueryDS.js (hunk 5 above)
Comment (greghutch) [reply components/bigqueryDS.js:207]: removed

File: components/bigqueryDS.js (hunk 6 above)
Comment (greghutch) [reply components/bigqueryDS.js:214]: Done.

**Pull request merged**